### DISCUSSION TITLE
Parsing error on SET (sqlite database)

### DISCUSSION THREAD

**Discussion by SebastiendOrnano** [sqlpage/SQLPage#644] [Q&A] [answered]
Sqlpage is a super app for no developer  low low coder like me !

I try to setup a simple website and I am following the documentation step by step. From yesterday I have a weird issue : my code at the beginning of a shell component was working and now I have parsing error : 
`SET $role = (
    SELECT role FROM users
    INNER JOIN sessions ON users.user_id = sessions.user_id
    WHERE sessions.session_id = sqlpage.cookie('session_token')
); 
`
`SET $role = (
         ⬆️
    SELECT role FROM users

sql parser error: Expected: identifier, found: $role at Line: 1, Column: 5`

I have done modification , with $ and withnot..... but I have still the parsing error. I have this SET instruction in another file with the same issue (working last week and no more working from yesterday).
I have install the 0.29 version yesterday so I tried with an older version 0.25 with the same  result ; Ublock is unlocked ; I emptied the cache between trials. It must be a very basic beginner coding error, but I can't find it !

Any help is welcome !

**Comment by lovasoa** [comment, marked as answer]
role is a reserved keyword in sql. Precious versions of our sql parser allowed it as a variable name, but newer versions are stricter. You can rename the variable to something like user_role

**Reply by SebastiendOrnano** [reply]
It works ! Thanks

**Comment by lovasoa** [comment]
I reported the issue upstream: https://github.com/apache/datafusion-sqlparser-rs/issues/1469

And proposed a fix : https://github.com/apache/datafusion-sqlparser-rs/pull/1474

Now just waiting for it to get merged

**Comment by lovasoa** [comment]
My fix was accepted, it'll be in the next version of SQLPage
